docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/ProcessMediaTrait.php

```diff
@@ -287,8 +290,9 @@ public function downloadFile($filename, $source_base_path, $drupal_file_director
    * @throws \Drupal\Core\Entity\EntityStorageException
    */
   public function createMediaEntity($fid, array $meta, $owner_id = 1) {
+    $file_manager = $this->entityTypeManager->getStorage('file');
     /** @var \Drupal\file\FileInterface $file */
-    $file = $this->entityTypeManager->getStorage('file')->load($fid);
+    $file = $file_manager->load($fid);
```

Comment:
We can avoid having to set `$file_manager` (and then later unset it) by wrapping loading the `$file` in a try/catch statement. Ditto the `$media_manager` variable later on.

Reply:
I see now that this is how it was working before, but without the try/catch block.

Reply:
Interesting! Didn't know that about the try/catch approach. I'll take a look at that when I get back to this.

Reply:
The try/catch block just accounts for the potential of an exception. Ideally we would code around all pieces of code that can throw exceptions with try/catch blocks (if there isn't already one in the method being called). In this case it allows us to chain the `load` method off of the `getStorage` method without having to worry about the fact that we might have gotten an exception from `getStorage`.